Synapse.RestClient/SynapseBaseApiClient.cs

```diff
@@ -57,7 +59,9 @@ protected T ParseResponse<T>(IRestResponse resp)
             if (resp.StatusCode != System.Net.HttpStatusCode.OK)
             {
                 var error = JsonConvert.DeserializeObject<SynapseApiErrorResponse>(resp.Content, this._jsonSettings);
-                throw new SynapseApiErrorException(error);
+                var ex = new SynapseApiErrorException(error);
+                this._apiErrorLogger?.Invoke(ex);
```

Comment:
Should check for null here. It could be passed in as null through the constructor.